### PR TITLE
Fix input checking for "Force night skip" (#1964)

### LINKED ISSUE #1964: A player could force night skip (bad user input checking)
beds/functions.lua allows players to force the night skip even if not enough players are in their beds.

The night skip can be forced by any player because the `fields.force` field is accepted from the user input irregardless whether enough players are in beds or not. A modified client could just send formspec data with the `fields.force` data set to the server and the night will be skipped.

https://github.com/minetest/minetest_game/blob/master/mods/beds/functions.lua#L216

### PR DESCRIPTION
**Pull request by beyondlimits**
Ugh, sooo... this fixes #1964 I think.

### CODE REVIEW

**Comment by paramat** [comment]
@sfan5 @sofar

File: mods/beds/functions.lua [hunk 1]
```diff
@@ -219,14 +219,22 @@ minetest.register_on_player_receive_fields(function(player, formname, fields)
...
+
 	if fields.quit or fields.leave then
 		lay_down(player, nil, nil, false)
 		update_formspecs(false)
 	end
 
 	if fields.force then
 		update_formspecs(is_night_skip_enabled())
-		if is_night_skip_enabled() then
+		local ges = #minetest.get_connected_players()
```
Comment (paramat) [inline mods/beds/functions.lua]: No need for 'ges', can be combined into following line and is actually clearere that way.

Comment (beyondlimits) [reply mods/beds/functions.lua]: Would probably follow if the whole code was *clear*.

**Comment by beyondlimits** [comment]
@paramat, I think you're right. I missed the sfan5's code though.

If I analyzed the code correctly, when player clicks "Force night skip" button, `update_formspecs(is_night_skip_enabled())` would show player a "Good morning" formspec (the one without "Force night skip" button), but further code might reject the operation.

**Comment by paramat** [comment]
Seems ok to sofar http://irc.minetest.net/minetest-dev/2018-07-03#i_5350400

**Pull request merged**